### PR TITLE
[Feature#1343][Connectors] add plugin: seatunnel-transform-spark-sleep

### LINKED ISSUE #1343: [Feature][transform] add a sleep transform plugin
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

add a sleep transform plugin for debug purpose.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yx91490**
## Purpose of this pull request

close #1343 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-transforms/seatunnel-transform-spark-sleep/src/main/scala/org/apache/seatunnel/spark/transform/Sleep.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    if (!result.isSuccess) {
+      return result
+    }
+    if (config.hasPath(KEY_TIMEUNIT)) {
+      val timeunit = config.getString(KEY_TIMEUNIT)
+      if (!VALID_TIMEUNITS.contains(timeunit)) {
+        CheckResult.error(s"TimeUnit '$timeunit' is invalid!")
+      }
+    }
+    CheckResult.success()
```
Comment (ruanwenjun) [inline seatunnel-transforms/seatunnel-transform-spark-sleep/src/main/scala/org/apache/seatunnel/spark/transform/Sleep.scala]: Do you aim to return an error result here? You need to add return in line 45 manually.

BTW, it's better to use if with return statement, this is more suitable in scala
```suggestion
if (config.hasPath(KEY_TIMEUNIT) && !VALID_TIMEUNITS.contains(config.getString(KEY_TIMEUNIT))) {
        CheckResult.error(s"TimeUnit '$timeunit' is invalid!")
} else {
    CheckResult.success()
}
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
There might be an error need to change here.

And I am not sure in which case the newly added transformer will be useful, since it doesn't output any information that would be helpful for debug.

**Comment by kyle-cx91** [comment]
> There might be an error need to change here.
> 
> And I am not sure in which case the newly added transformer will be useful, since it doesn't output any information that would be helpful for debug.

@yx91490 I'm interested in this feature too, could u please describe how you debug with this sleep plugin ?

File: seatunnel-transforms/seatunnel-transform-spark-sleep/src/main/scala/org/apache/seatunnel/spark/transform/Sleep.scala (hunk 1 above)
Comment (yx91490) [reply seatunnel-transforms/seatunnel-transform-spark-sleep/src/main/scala/org/apache/seatunnel/spark/transform/Sleep.scala]: sorry, very thanks!

**Comment by yx91490** [comment]
> > There might be an error need to change here.
> > And I am not sure in which case the newly added transformer will be useful, since it doesn't output any information that would be helpful for debug.
> 
> @yx91490 I'm interested in this feature too, could u please describe how you debug with this sleep plugin ?

you can use your imagination :)

**Comment by liujinhui1994** [comment]
@yx91490 Can you help me understand what scenario this transform is applied to?

**Comment by EricJoy2048** [comment]
What's the news?

It's a long time since the last comment. If there is on further change, I suggest closing this pr.

**Comment by EricJoy2048** [comment]
I close this PR because there was no feedback for a long time.

**Pull request closed without merging**